### PR TITLE
Avoid continuous rendering between video captures

### PR DESCRIPTION
**Pull request by AntoineRichard**
# Description

`--video` enabled a headless Kit visualizer and continuous PhysX Fabric transform updates for the full training run, even though frames are only needed during capture windows. This caused a substantial FPS drop between recordings.

This change treats headless visualizers as on-demand rather than continuous renderers, disables continuous Fabric synchronization after PhysX initialization, and synchronizes transforms immediately before each visualizer frame capture. RTX camera sensors explicitly retain continuous Fabric updates.

### RTX 5090 benchmark

`Isaac-Humanoid-Direct`, PhysX, 4,096 environments, RSL-RL, 16 iterations. Idle results exclude iterations 0 and 10, which contain the two scheduled 32-frame capture windows.

| Configuration | Mean idle FPS | Difference |
| --- | ---: | ---: |
| Plain headless | 360,635 | — |
| Offscreen RTX enabled, no video recorder | 326,336 | -9.5% vs. plain headless |
| Periodic `--video` with this fix | 315,966 | -3.2% vs. offscreen RTX control |

The unfixed periodic-video run reproduced approximately 214,300 FPS between captures, about 34% below its plain-headless control. With this fix, most of that avoidable idle-window regression is removed. Capture iterations still drop as expected; the fixed periodic run measured 76,127 and 135,151 FPS during its two capture iterations, with 289,675 FPS overall including both captures.

No new dependencies are required.

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## Release backport

- [ ]  Backport this pull request to the active release branch after it merges into `develop`

## Screenshots

Not applicable; this is a performance fix. A frame probe on the RTX 5090 verified that all 20 captured frames were non-black and changed over time (`temporal_std=72.21`, mean consecutive delta `12.46`).

## Checklist

Docker and GPU tests run on demand. Push the commits you want tested, then
comment `run-ci` on the pull request.

- [x] I have read and understood the [contribution guidelines](https://isaac-sim.github.io/IsaacLab/main/source/refs/contributing.html)
- [x] I have run the [`pre-commit` checks](https://pre-commit.com/) with `./isaaclab.sh --format`
- [x] I have made corresponding changes to the documentation (not applicable; no public API or user workflow changed)
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added a changelog fragment under `source/<pkg>/changelog.d/` for every touched package (do **not** edit `CHANGELOG.rst` or bump `extension.toml` — CI handles that)
- [x] I have added my name to the `CONTRIBUTORS.md` or my name already exists there

### Validation

- `uv run --frozen --extra test --extra rerun python -m pytest source/isaaclab/test/sim/test_simulation_context_visualizers.py source/isaaclab/test/envs/test_video_recorder.py -q` — 74 passed
- `uv run --frozen isaaclab -f` — all hooks passed
- `source/isaaclab/test/sensors/test_camera.py::test_camera_init` — passed on RTX 5090

### CODE REVIEW

File: source/isaaclab/isaaclab/sensors/camera/camera.py [hunk 1]
```diff
@@ -198,6 +198,7 @@ def __init__(self, cfg: CameraCfg):
 
             settings = get_settings_manager()
             settings.set_bool("/isaaclab/render/rtx_sensors", True)
+            settings.set_bool("/physics/fabricUpdateTransformations", True)
```
Comment (kellyguo11) [inline source/isaaclab/isaaclab/sensors/camera/camera.py:201]: why is this needed here?

File: source/isaaclab/isaaclab/envs/utils/video_recorder.py [hunk 2]
```diff
@@ -196,6 +196,8 @@ def _frame_from_visualizer(self, viz_type: str, sub: str) -> np.ndarray | None:
                 )
 
         viz = candidates[0]
+        if not sim.is_rendering:
+            sim.forward()
```
Comment (kellyguo11) [inline source/isaaclab/isaaclab/envs/utils/video_recorder.py:200]: should this be the responsibility of the video recorder?

**Review by kellyguo11 — COMMENTED** [review]
@matthewtrepte for review

**Comment by matthewtrepte** [comment]
ah nice find Antoine. 

for an edge case, can you test if we launch multiple headless kit video recordings at once (with multiple VideoRecordingCfg files), if the on demand method doesn't cause any issues.

File: source/isaaclab/isaaclab/sensors/camera/camera.py (hunk 1 above)
Comment (AntoineRichard) [reply source/isaaclab/isaaclab/sensors/camera/camera.py:201]: SimulationContext sets `/physics/fabricUpdateTransformations` during initialization, before scene-owned Camera objects are constructed. An Isaac RTX camera subsequently makes `is_rendering` true by setting the adjacent `rtx_sensors` flag, but that property does not mutate the Fabric setting. Without this line, Fabric remains disabled and camera transforms can be stale. Camera already owns the RTX sensor flag for the same lifecycle reason; the TODO above tracks moving both settings to a renderer pre-reset hook. `test_camera_init` verifies this path on the RTX 5090.

File: source/isaaclab/isaaclab/envs/utils/video_recorder.py (hunk 2 above)
Comment (AntoineRichard) [reply source/isaaclab/isaaclab/envs/utils/video_recorder.py:200]: Yes—the current placement in `VideoRecorder` is intentional. It is the component that knows an on-demand frame is about to be requested. Moving the synchronization into `KitVisualizer` would couple the visualizer to the active physics backend and would also affect non-recorder consumers. `sim.forward()` delegates to the configured `PhysicsManager` and only runs here when continuous rendering is disabled. The dual-recorder integration test added in d319ed5b8ef exercises this boundary with two simultaneous capture requests.

**Comment by AntoineRichard** [comment]
@matthewtrepte I tested one explicit `KitVisualizerCfg(headless=True)` shared by two simultaneous `VideoRecorderCfg(source="visualizer:kit")` entries. Both captured 20/20 non-black, moving frames on the RTX 5090; temporal standard deviations were 37.50 and 37.43. I added the GPU integration coverage in d319ed5b8ef. It mocks only the MoviePy encoding boundary and exercises the real PhysX and headless Kit capture path.

**Review by matthewtrepte — APPROVED** [review]

**Review by StafaH — APPROVED** [review]

**Comment by AntoineRichard** [comment]
run-ci

**Comment by AntoineRichard** [comment]
run-ci
